exec/java-exec/src/main/java/org/apache/drill/exec/service/ServiceEngine.java

```diff
@@ -177,6 +182,13 @@ public void run() {
     });
   }
 
+  /** If in distributed mode, disallow using loopback address to register new drillbit. */
+  private void checkLoopbackAddress(String address) throws DrillbitStartupException, UnknownHostException {
+    if (isDistributedMode && InetAddress.getByName(address).isLoopbackAddress()) {
+      throw new DrillbitStartupException("Loopback address is not allowed to be registered in distributed mode");
```

Comment:
How about "Drillbit is disallowed to bind to loopback address in distributed mode."?